core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -332,39 +350,26 @@ public boolean isPreloaded() {
 
    @Override
    public CompletionStage<Void> preload() {
-      if (!enabled)
-         return CompletableFutures.completedNull();
-
-      AdvancedCacheLoader<Object, Object> preloadCl = null;
-
-      acquireReadLock();
-      try {
-         for (CacheLoader l : loaders) {
-            if (getStoreConfig(l).preload()) {
-               if (!(l instanceof AdvancedCacheLoader)) {
-                  throw new PersistenceException("Cannot preload from cache loader '" + l.getClass().getName()
-                        + "' as it doesn't implement '" + AdvancedCacheLoader.class.getName() + "'");
-               }
-               preloadCl = (AdvancedCacheLoader) l;
-               if (preloadCl instanceof AdvancedAsyncCacheLoader)
-                  preloadCl = (AdvancedCacheLoader) ((AdvancedAsyncCacheLoader) preloadCl).undelegate();
-               break;
-            }
-         }
-      } finally {
-         releaseReadLock();
-      }
-      if (preloadCl == null) {
+      long stamp = acquireReadLock();
+      NonBlockingStore<Object, Object> nonBlockingStore = getStoreLocked(storeStatus ->
+            storeStatus.config.preload());
```

Comment:
Nitpick: `s/storeStatus/store` and it will easily fit on one line.

Reply:
Sure, I will just change to status.